### ISSUE TITLE
[LXD] Container initialization with command/script

### ISSUE THREAD

**Issue by Davide-DD** [eclipse/fog05#125]
**Is your feature request related to a problem? Please describe.**
I have multiple containers and I need to execute a script in each one of them once they're started. 

**Describe the solution you'd like**
I'm trying to use cloud-init initialization that I saw on LXD_plugin, but it seems it hasn't been fully implemented yet: I have added the object "configuration" to the FDU descriptor, but it seems it's not taken into consideration when parsing the descriptor. 

**Describe alternatives you've considered**
I'm using cron for now, but it would be extra nice to have a feature like this fully working, especially in development/testing environment, when it's needed to change fast what is being executed.

**Comment by gabrik** [comment]
Thanks @Davide-DD, actually the cloud-init is supposed to work.
We will do more testing to see if something has messed up with this feature.

Another workaround, more elegant than using cronjobs,  can be to setup a service (using systemd or other init system) in the image when preparing it.

But cloud-init is supposed to work

**Comment by gabrik** [comment]
After some investigation seems that:
- LXD plugin was overwriting the cloud-init configuration
- If confiugration is passed to the container, if the container is missing cloud-init it is ignored

We are fixing the problem with the LXD plugin, but the second "issue" is related to the base image of the container, a solution can be enable also other remotes for the LXD plugin, as the plain LXD image repository does not contain images with cloud-init installed, while for instance the ubuntu image repository contains images with cloud-init installed.

A solution can be add the possibility to use also the ubuntu repository with an URI like `ubuntu://<ubuntu version>`

I keep this open until the bug in LXD plugin is fixed

**Comment by gabrik** [comment]
Solved by #128 
Example of FDU with Ubuntu container and simple cloud-init script:

```
{
    "uuid": "1b526aa7-408f-4d63-b58c-9e2dfd9451ed",
    "name": "test2if",
    "configuration": {
        "conf_type": "CLOUD_INIT",
        "script": "#cloud-config\nruncmd:\n  - [touch, /tmp/cinit_was_here]"
    },
    "computation_requirements": {
        "cpu_arch": "x86_64",
        "cpu_min_freq": 0,
        "cpu_min_count": 1,
        "ram_size_mb": 128.0,
        "storage_size_gb": 5.0
    },
    "image": {
        "uri": "ubuntu://bionic",
        "checksum": "",
        "format": ""
    },
    "hypervisor": "LXD",
    "migration_kind": "LIVE",
    "depends_on": [],
    "interfaces": [
        {
            "name": "eth0",
            "is_mgmt": true,
            "if_type": "EXTERNAL",
            "mac_address": "be:ef:be:ef:00:02",
            "virtual_interface": {
                "intf_type": "BRIDGED",
                "vpci": "lxdbr0",
                "bandwidth": 10
            },
            "cp_id": ""
        }
    ],
    "io_ports": [],
    "connection_points": []
}
```
